packages/@aws-cdk/core/lib/deps.ts

```diff
@@ -2,8 +2,14 @@ import { CfnResource } from './cfn-resource';
 import { Stack } from './stack';
 import { Stage } from './stage';
 import { findLastCommonElement, pathToTopLevelStack as pathToRoot } from './util';
+import { IConstruct } from 'constructs';
 
-type Element = CfnResource | Stack;
+/**
+ * This interface is intended to represent an object that is either a Stack or a CfnResource
+ */
+export interface IElement extends IConstruct {
```

Comment:
No need for empty interfaces...can we keep the original type `Element` here?

Reply:
@TheRealAmazonKendra did not like the `Element` type, since it is a union type and doesn't play well with jsii. At her request, I went about removing it where I could.

Reply:
@TheRealAmazonKendra I think the union type `Element` is safe here, since it isn't exported; since we already had the `Element` type (and it provides better safety than just `IConstruct`, I'd prefer to keep it.

Reply:
Happy to do any way you all decide but FYI:
> So the Element or IElement ends up being part of the public interface as the return type of CfnResource.obtainDependencies(), so that's why I moved away from the union type all the way back into deps.ts.
> 
> So far as the benefit of IElement over just accepting IConstruct, there isn't one as written. I added this to be a placeholder for a possibly more specific interface after feedback from AWS, since I think we may wish to polish that piece up a touch more. This interface is what I was referring to when I asked for help at the end of this conversation:
https://github.com/aws/aws-cdk/pull/20419#discussion_r944611432

[From down here](https://github.com/aws/aws-cdk/pull/20419#discussion_r1025747070)